applications/serial_lte_modem/src/gnss/Kconfig

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+config SLM_GNSS
+	bool "GNSS support in SLM"
+	default y
+
+if SLM_GNSS
+
+config SLM_AGPS
+	bool "nRF Cloud A-GPS support in SLM"
+	select CLOUD_API
+	select MODEM_INFO
+	select NRF_CLOUD
+	select NRF_CLOUD_AGPS
+	help
+	  Enable nRF Cloud A-GPS for GPS
```

Comment:
We should generally be careful on such select, as what they select will stay selected even when outer symbol is de-selected.

In this case, the risk is smaller because there is no `default y` here, but if `SLM_AGPS=y` is changed into `SLM_AGPS=n` using menuconfig, then the symbols `CLOUD_API`, `MODEM_INFO`, `NRF_CLOUD`, and `NRF_CLOUD_AGPS` stays selected.
See more: https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

Not sure I understand the need for constantly wrapping other Kconfig symbols.
It makes it harder for our users to understand that what the really need to enable is: `NRF_CLOUD_AGPS`.

If you need the symbol available at higher level, you can do so, like:
```
config NRF_CLOUD_AGPS
	bool " Enable nRF Cloud Assisted GPS (NEW)"
```
That will just give you the symbol in two places, and then let the use reconfigure `NRF_CLOUD_AGPS` either here, or in its original location, no reason for wrapping the original  symbol inside a new symbol.

Alternatively, if you really want to wrap those symbols inside new names, then safeguard the original symbols, as was discussed and done in this PR: https://github.com/nrfconnect/sdk-nrf/pull/4546

-------

Secondly, could you improve the help text ?
You have a very nice decription here:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/nrf9160/agps/README.html

but for users just opening a sample, it is always nice to get some more info, for example:
```
	help
	  Enable nRF Cloud Assisted GPS (A-GPS). 
	  The use of A-GPS reduces the time for a GPS device to estimate its position.
	  It uses nRF Connect for Cloud A-GPS service for faster download of broadcasted satellite information.
```

Reply:
Thanks, I removed all SELECT in KCONFIG and set those needed in prj.conf.
For help text, please allow me to keep it short.
For other comments, please help to review again.

Reply:
> Thanks, I removed all SELECT in KCONFIG and set those needed in prj.conf.

Just removing the `select` and not having something else in its place is not a solution.
Doing so allows the user to, for example deselect `MODEM_INFO` which will result in linking errors like this:
```
/opt/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld: app/libapp.a(slm_at_gnss.c.obj): in function `cell_pos_req_wk':
/projects/github/ncs/nrf/applications/serial_lte_modem/build/../src/gnss/slm_at_gnss.c:128: undefined reference to `nrf_cloud_cell_pos_request'
/opt/gcc-arm-none-eabi-9-2020-q2-update/bin/../lib/gcc/arm-none-eabi/9.3.1/../../../../arm-none-eabi/bin/ld: app/libapp.a(slm_at_gnss.c.obj): in function `pgps_req_wk':
/projects/github/ncs/nrf/applications/serial_lte_modem/build/../src/gnss/slm_at_gnss.c:110: undefined reference to `nrf_cloud_pgps_request_all'
```

>  For help text, please allow me to keep it short.

Why ?
Help texts are important for users, so what is the value in keeping it short.

>  For other comments, please help to review again.

let's discuss directly.